system/Database/BaseBuilder.php

```diff
@@ -153,6 +153,13 @@ class BaseBuilder
      */
     protected $QBIgnore = false;
 
+    /**
+     * QB Options data
+     *
+     * @var array
+     */
+    protected $QBOptions;
```

Comment:
Why is the property named `QBOptions`, not `QBUpsert`?

Reply:
I had contemplated adding a parameter to the $renderMethod functions in executeBatch(). This would add $options parameter to _insertBatch() and _upsertBatch(). However I found it easier just to access the property from those methods rather than pass it to them. 

Additionally I want to add the method fromQuery() as I demonstrated in the comments here. I would use the $QBOptions to hold this query would could be used for both inserts and upserts. So I didn't want to associate it with upsert. 

I could of created unique properties for all these things but I felt it would clutter the code unnecessarily.